packages/repository-tests/src/crud/transactions.suite.ts

```diff
@@ -71,12 +71,16 @@ export function transactionSuite(
           tx = undefined;
         });
         afterEach(async () => {
-          // FIXME: replace tx.connection with tx.isActive when it becomes
-          // available
-          // see https://github.com/strongloop/loopback-next/issues/3471
-          tx = setTransactionToUndefined(tx);
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          if (tx !== undefined && (tx as any).connection) {
+          // For postgresql connector, it doesn't wrap the connection with
+          // the Transaction that has juggler's Transaction mixin applied,
+          // therefore `commit` or `rollback` won't delete the connection's
+          // reference of a `tx`.
+          // Detailed explanation see:
+          // https://github.com/strongloop/loopback-next/pull/4474
+          if (ds.connector && ds.connector.name === 'postgresql') {
+            tx = undefined;
+          }
+          if (tx?.isActive()) {
```

Comment:
This test will be always `true` if `tx.isActive()` returns a promise.

Reply:
@raymondfeng good catch!
it is not a promise, fixed.